### PR TITLE
const std::string& in KeyExprView

### PR DESCRIPTION
**Pull request by milyin**
fix for https://github.com/eclipse-zenoh/zenoh-cpp/issues/49

### CODE REVIEW

File: examples/simple/universal/z_simple.cxx [hunk 1]
```diff
@@ -30,24 +30,26 @@ class CustomerClass {
     CustomerClass& operator=(const CustomerClass&) = delete;
     CustomerClass& operator=(CustomerClass&&) = delete;
 
-    CustomerClass() : session(nullptr), pub(nullptr) {
+    CustomerClass(const KeyExprView& keyexpr) : session(nullptr), pub(nullptr) {
         Config config;
         Session s = std::get<Session>(open(std::move(config)));
```
Comment (p-avital) [inline examples/simple/universal/z_simple.cxx:35]: Not really the PR's goal, but could we avoid having examples where we encourage users to construct `Session`s in classes? It's literally the worst thing they could do.

**Review by p-avital — CHANGES_REQUESTED** [review]

File: examples/simple/universal/z_simple.cxx (hunk 1 above)
Comment (milyin) [reply examples/simple/universal/z_simple.cxx:35]: Will do it in separate PR

Comment (milyin) [reply examples/simple/universal/z_simple.cxx:35]: https://github.com/eclipse-zenoh/zenoh-cpp/issues/51

**Review by milyin — COMMENTED** [review]
Changes to be made in separate PR, as fix to https://github.com/eclipse-zenoh/zenoh-cpp/issues/51

**Pull request merged**